src/messages.js

```diff
@@ -65,9 +72,111 @@
+      JSAV.ext.textToSpeech(txt);
+      // SwitchingOff sound buttons in other frames
+      for (var j = 0; j < window.length; j++) {
+        $(".sound", window.frames[j].document).not(this).each(function() {
+          $(this.offsetParent).parent().find(".jsavcontainer").attr("voice", "false");
+          $(this).removeClass("sound").addClass("soundOff");
+        });
+      }
+      // SwitchingOff sound buttons not in frames
+      $(".sound", window.document).not(this).each(function() {
+        $(this.offsetParent).parent().find(".jsavcontainer").attr("voice", "false");
+        $(this).removeClass("sounds").addClass("soundOff");
+      });
+      //Toggle voice attribute and stop the current voice
+      var synth = window.speechSynthesis;
+      var amISpeaking = synth.speaking;
+      // "inlineav" slideshows
+      if($(this.offsetParent).attr("voice") !== undefined) {
+        if($(this.offsetParent).attr("voice") === "true") {
+          $(this.offsetParent).attr("voice", "false");
+          if(amISpeaking === true) {
+            synth.cancel();
+          }
+        } else {
+          $(this.offsetParent).attr("voice", "true");
+        } //slideshows in a frame
+      } else if($(this.offsetParent).parent().find(".jsavcontainer").attr("voice") === "true") {
+        $(this.offsetParent).parent().find(".jsavcontainer").attr("voice", "false");
+        if(amISpeaking === true) {
+          synth.cancel();
+        }
+      } else {
+        $(this.offsetParent).parent().find(".jsavcontainer").attr("voice", "true");
+      }
+      //Toggle button class sound/soundOff
+      if($(this).hasClass("sound")) {
+        $(this).removeClass("sound").addClass("soundOff");
+        jsav.logEvent({
+          "type": "jsav-narration-off",
+          "currentStep": jsav.currentStep(),
+          "totalSteps": jsav.totalSteps()
+        });
+      } else {
+        $(this).removeClass("soundOff").addClass("sound");
+        jsav.logEvent({
+          "type": "jsav-narration-on",
+          "currentStep": jsav.currentStep(),
+          "totalSteps": jsav.totalSteps()
+        });
+      }
+    });
+    return $elem;
+  };
+/////////end of sound control
+/////////////////////////////////////////
   JSAV.init(function(options) {
+    var self = this;
     var output = options.output ? $(options.output) : $(this.container).find(".jsavoutput");
     this._msg = new MessageHandler(this, output);
+    if (options.narrationEnabled === true) {
+      //adding sound button if there is no sound button in the container and the container has both controls and settings button
+      if(($(this.container).parent().find(".new").length === 0) && ($(this.container).find(".jsavcontrols").length !== 0) && ($(this.container).parent().find(".jsavsettings").length !== 0)) {
+        $(this.container).parent().find(".jsavsettings").wrap("<span class='new'></span>");
+        $(this.container).parent().find(".new").append(JSAV.ext.soundSettings(self));
```

Comment:
There's no need for self here, you can just use `this` (and remove the self=this from above). Also, can use `this.soundSettings()`.